### PR TITLE
Add new extra check queries

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### New queries added:
- [x] iam_access_analyzer_enabled_without_findings
- [x] apigateway_rest_api_endpoint_configured_to_public
- [x] api_gatewayv2_route_authorization_configured
- [x] backup_plan_region_configured
- [x] backup_vault_region_configured
- [x] cloudfront_distributions_field_level_encryption_enabled
- [x] cloudtrail_insights_exist
- [x] ec2_ami_publicly_accessible
- [x] ebs_snapshot_encrypted
- [x] drs_job_exists 
- [x] vpc_security_group_allows_ingress_to_mongodb_ports
- [x] vpc_security_group_allows_ingress_to_cassandra_ports
- [x] vpc_security_group_allows_ingress_to_oracle_ports
- [x] vpc_security_group_allows_ingress_to_memcached_port
- [x] guardduty_no_high_severity_findings

### CODE REVIEW

File: conformance_pack/apigateway.sp [hunk 1]
```diff
@@ -252,6 +272,43 @@ query "apigateway_rest_api_authorizers_configured" {
   EOQ
 }
 
+query "apigateway_rest_api_endpoint_configured_to_public" {
+  sql = <<-EOQ
+    select
+      name as resource,
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:278]: can we please construct arn ?

File: conformance_pack/apigateway.sp [hunk 2]
```diff
@@ -252,6 +272,43 @@ query "apigateway_rest_api_authorizers_configured" {
...
+      case
+        when endpoint_configuration_types ? 'PRIVATE' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when endpoint_configuration_types ? 'PRIVATE' then name || ' configured to private endpoint.'
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:284]:
```suggestion
        when endpoint_configuration_types ? 'PRIVATE' then name || ' not publicly accessible.'
```

File: conformance_pack/apigateway.sp [hunk 3]
```diff
@@ -252,6 +272,43 @@ query "apigateway_rest_api_authorizers_configured" {
...
+        else name || ' configured to public endpoint.'
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:285]:
```suggestion
        else name || ' publicly accessible.'
```

File: conformance_pack/apigateway.sp [hunk 4]
```diff
@@ -109,6 +109,26 @@ control "apigateway_rest_api_authorizers_configured" {
...
+  description   = "This control checks whether API Gateway endpoint is public or private. This rule is non compliant if API Gateway endpoint is public."
+  query         = query.apigateway_rest_api_endpoint_configured_to_public
+
+  tags = merge(local.conformance_pack_apigateway_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "api_gatewayv2_route_authorizer_enabled" {
+  title         = "Ensure that API Gateway V2  has authorizer configured"
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:123]:
```suggestion
  title         = "Ensure that API Gateway V2 has authorizer configured"
```

File: conformance_pack/apigateway.sp [hunk 5]
```diff
@@ -109,6 +109,26 @@ control "apigateway_rest_api_authorizers_configured" {
...
+  description   = "This control checks whether API Gateway V2  has authorizer configured. This rule is non compliant if API Gateway V2 have no authorizers configured."
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:124]:
```suggestion
  description   = "This control checks whether API Gateway V2 has authorizer configured. This rule is non compliant if API Gateway V2 have no authorizers configured."
```

File: conformance_pack/cloudfront.sp [hunk 6]
```diff
@@ -148,6 +148,16 @@ control "cloudfront_distribution_origin_access_identity_enabled" {
   })
 }
 
+control "cloudfront_distributions_field_level_encryption_enabled" {
```
Comment (khushboo9024) [inline conformance_pack/cloudfront.sp:151]:
```suggestion
control "cloudfront_distribution_field_level_encryption_enabled" {
```

File: conformance_pack/cloudfront.sp [hunk 7]
```diff
@@ -148,6 +148,16 @@ control "cloudfront_distribution_origin_access_identity_enabled" {
...
+  title       = "CloudFront distributions have Field Level Encryption enabled"
```
Comment (khushboo9024) [inline conformance_pack/cloudfront.sp:152]:
```suggestion
  title       = "CloudFront distributions should have field level encryption enabled"
```

File: conformance_pack/ebs.sp [hunk 8]
```diff
@@ -141,6 +141,16 @@ control "ebs_volume_unused" {
   })
 }
 
+control "ebs_snapshot_encrypted" {
```
Comment (khushboo9024) [inline conformance_pack/ebs.sp:144]:
```suggestion
control "ebs_snapshot_encryption_enabled" {
```

File: conformance_pack/ec2.sp [hunk 9]
```diff
@@ -269,6 +269,16 @@ control "ec2_instance_virtualization_type_no_paravirtual" {
   })
 }
 
+control "ec2_ami_publicly_accessible" {
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:272]:
```suggestion
control "ec2_ami_restrict_public_access" {
```

File: conformance_pack/ec2.sp [hunk 10]
```diff
@@ -269,6 +269,16 @@ control "ec2_instance_virtualization_type_no_paravirtual" {
...
+  title       = "Ensure there are no EC2 AMIs set as Public"
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:273]:
```suggestion
  title       = "Ensure there are no EC2 AMIs set as public"
```

File: conformance_pack/ec2.sp [hunk 11]
```diff
@@ -269,6 +269,16 @@ control "ec2_instance_virtualization_type_no_paravirtual" {
...
+  description = "This control checks whether EC2 AMIs are set as Private or not. The control fails if the EC2 AMIs are set as Public."
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:274]:
```suggestion
  description = "This control checks whether EC2 AMIs are set as private or not. The control fails if the EC2 AMIs are set as public."
```

File: conformance_pack/iam.sp [hunk 12]
```diff
@@ -1376,6 +1386,29 @@ query "iam_policy_unused" {
...
+      case
+        when aa.status = 'ACTIVE' and aa.findings is null then 'IAM Access Analyzer does not have active findings.'
+        when aa.status = 'ACTIVE' and jsonb_array_length(aa.findings) > 0 then 'IAM Access Analyzer has active findings.'
+        when aa.status = 'NOT_AVAILABLE' then 'IAM Access Analyzer is not enabled.'
+        else 'IAM Access Analyzer is not active.'
+      end as reason
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "r.")}
+    from
+      aws_region as r
+      left join aws_accessanalyzer_analyzer as aa on r.account_id = aa.account_id and r.region = aa.region;
```
Comment (khushboo9024) [inline conformance_pack/iam.sp:1408]:
```
    select
      'arn:' || r.partition || '::' || r.region || ':' || r.account_id as resource,
      case
        -- Skip any regions that are disabled in the account.
        when r.opt_in_status = 'not-opted-in' then 'skip'
        when aa.status = 'ACTIVE' and aa.findings is null then 'ok'
        when aa.status = 'ACTIVE' and jsonb_array_length(aa.findings) > 0 then 'alarm'
        when aa.status = 'NOT_AVAILABLE' then 'alarm'
        else 'alarm'
      end as status,
      case
        when r.opt_in_status = 'not-opted-in' then r.region || ' region is disabled.'
        when aa.status = 'ACTIVE' and aa.findings is null then aa.name || ' does not have active findings in region ' || r.region || '.'
        when aa.status = 'ACTIVE' and jsonb_array_length(aa.findings) > 0 then aa.name || ' has active findings in region ' || r.region || '.'
        when aa.status = 'NOT_AVAILABLE' then aa.name || ' is not enabled in region ' || r.region || '.'
        else 'IAM Access Analyzer is not active in region ' || r.region || '.'
      end as reason
      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "r.")}
    from
      aws_region as r
      left join aws_accessanalyzer_analyzer as aa on r.account_id = aa.account_id and r.region = aa.region;
      ```

File: conformance_pack/cloudtrail.sp [hunk 13]
```diff
@@ -205,6 +205,16 @@ control "cloudtrail_multi_region_read_write_enabled" {
   })
 }
 
+control "cloudtrail_insights_exist" {
```
Comment (khushboo9024) [inline conformance_pack/cloudtrail.sp:208]:
```suggestion
control "cloudtrail_trail_insight_selectors_and_logging enabled" {
```

File: conformance_pack/cloudfront.sp [hunk 14]
```diff
@@ -148,6 +148,16 @@ control "cloudfront_distribution_origin_access_identity_enabled" {
   })
 }
 
+control "cloudfront_distribution_field_level_encryption_enabled" {
+  title       = "CloudFront distributions should have field level encryption enabled"
+  description = "This control checks whether an Amazon CloudFront distribution have Field Level Encryption enabled. The control fails if CloudFront distribution Field Level Encryption is not enabled."
```
Comment (khushboo9024) [inline conformance_pack/cloudfront.sp:153]:
```suggestion
  description = "This control checks whether an Amazon CloudFront distribution have field level encryption enabled. The control fails if CloudFront distribution field level encryption is not enabled."
```

File: conformance_pack/drs.sp [hunk 15]
```diff
@@ -0,0 +1,48 @@
+locals {
+  conformance_pack_drs_common_tags = merge(local.aws_compliance_common_tags, {
+    service = "AWS/DRS"
+  })
+}
+
+control "drs_job_exists" {
```
Comment (khushboo9024) [inline conformance_pack/drs.sp:7]:
```suggestion
control "drs_job_enabled" {
```

File: conformance_pack/drs.sp [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+      end as status,
+      case
+        when drs_job_count.count = 0 or drs_job_count.count is null then 'DRS job does not exist for ' || r.region || '.'
+        else 'DRS job exist for ' || r.region || '.'
+      end as reason
+    from
+      aws_region as r
+    left join drs_job_count
+    on
+      r.region = drs_job_count.region;
```
Comment (khushboo9024) [inline conformance_pack/drs.sp:46]:
```suggestion
      aws_region as r
      left join drs_job_count on r.region = drs_job_count.region;
```

File: conformance_pack/drs.sp [hunk 17]
```diff
@@ -0,0 +1,48 @@
+locals {
+  conformance_pack_drs_common_tags = merge(local.aws_compliance_common_tags, {
+    service = "AWS/DRS"
+  })
+}
+
+control "drs_job_exists" {
+  title       = "Ensure DRS is enabled with jobs"
+  description = "Check if DRS is enabled with jobs. This rule is non-compliant if DRS is not enabled with jobs for a particular region."
```
Comment (khushboo9024) [inline conformance_pack/drs.sp:9]:
```suggestion
  description = "Ensure if DRS is enabled with jobs. This rule is non-compliant if DRS is not enabled with jobs for a particular region."
```

File: conformance_pack/drs.sp [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+        _ctx
+    )
+    select
+      'arn:' || r.partition || '::' || r.region || ':' || r.account_id as resource,
+      case
+        when drs_job_count.count = 0 or drs_job_count.count is null then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when drs_job_count.count = 0 or drs_job_count.count is null then 'DRS job does not exist for ' || r.region || '.'
```
Comment (khushboo9024) [inline conformance_pack/drs.sp:39]:
```suggestion
        when drs_job_count.count = 0 or drs_job_count.count is null then 'DRS job not enabled for ' || r.region || '.'
```

File: conformance_pack/drs.sp [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+        else 'DRS job exist for ' || r.region || '.'
```
Comment (khushboo9024) [inline conformance_pack/drs.sp:40]:
```suggestion
        else 'DRS job enabled for ' || r.region || '.'
```

File: conformance_pack/ebs.sp [hunk 20]
```diff
@@ -141,6 +141,16 @@ control "ebs_volume_unused" {
   })
 }
 
+control "ebs_snapshot_encryption_enabled" {
+  title       = "EBS snapshots should be encrypted"
+  description = "Check if EBS snapshots are encrypted. This rule is non-compliant if the EBS snapshot is not encrypted."
```
Comment (khushboo9024) [inline conformance_pack/ebs.sp:146]:
```suggestion
  description = "Ensure if EBS snapshots are encrypted. This rule is non-compliant if the EBS snapshot is not encrypted."
```

File: conformance_pack/guardduty.sp [hunk 21]
```diff
@@ -50,6 +50,16 @@ control "guardduty_finding_archived" {
   })
 }
 
+control "guardduty_no_high_severity_findings" {
+  title       = "GuardDuty Detector should not have high severity finding."
```
Comment (khushboo9024) [inline conformance_pack/guardduty.sp:54]:
```suggestion
  title       = "GuardDuty Detector should not have high severity finding"
```

File: conformance_pack/guardduty.sp [hunk 22]
```diff
@@ -77,6 +87,37 @@ query "guardduty_enabled" {
...
+        when fc.count = 0  or fc.count is NULL then d.detector_id || ' is enabled and does not have high severity findings.'
+        else d.detector_id || ' is enabled and has ' || fc.count ||' high severity findings.'
+      end as reason
+      ${replace(local.tag_dimensions_qualifier_sql, "__QUALIFIER__", "d.")}
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "d.")}
+    from
+      aws_guardduty_detector as d
+      left join finding_count as fc on fc.detector_id = d.detector_id
+    where 
+      status = 'ENABLED';
```
Comment (khushboo9024) [inline conformance_pack/guardduty.sp:117]: can we remove this where clause and add one condition to skip the control if `status <>  'ENABLED`?

File: conformance_pack/iam.sp [hunk 23]
```diff
@@ -1376,6 +1386,33 @@ query "iam_policy_unused" {
...
+        when aa.status = 'ACTIVE' and aa.findings is null then aa.name || ' does not have active findings in region ' || r.region || '.'
+        when aa.status = 'ACTIVE' and jsonb_array_length(aa.findings) > 0 then aa.name || ' has active findings in region ' || r.region || '.'
+        when aa.status = 'NOT_AVAILABLE' then aa.name || ' is not enabled in region ' || r.region || '.'
+        else 'IAM Access Analyzer is not active in region ' || r.region || '.'
+      end as reason
+      ${replace(local.common_dimensions_qualifier_sql, "__QUALIFIER__", "r.")}
+    from
+      aws_region as r
+      left join aws_accessanalyzer_analyzer as aa on r.account_id = aa.account_id and r.region = aa.region;
+      ```
```
Comment (khushboo9024) [inline conformance_pack/iam.sp:1412]:
```suggestion
```

File: conformance_pack/vpc.sp [hunk 24]
```diff
@@ -969,6 +1009,179 @@ query "vpc_configured_to_use_vpc_endpoints" {
...
+        type = 'ingress'
+        and cidr_ipv4 = '0.0.0.0/0'
+        and (
+            ( ip_protocol = '-1'
+            and from_port is null
+            )
+            or (
+                from_port >= 27017
+                and to_port <= 27018
+            )
```
Comment (khushboo9024) [inline conformance_pack/vpc.sp:1030]:
```suggestion
            or (
                from_port >= 27017
                and to_port <= 27017
            )
            or (
                from_port >= 27018
                and to_port <= 27018
            )
```

**Comment by khushboo9024** [comment]
@Priyanka-Chatterjee-2000 Please check below error.
image

File: conformance_pack/ebs.sp [hunk 25]
```diff
@@ -317,3 +327,22 @@ query "ebs_volume_unused" {
...
+query "ebs_snapshot_encryption_enabled" {
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when encrypted then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when encrypted then title || ' is encrypted.'
```
Comment (khushboo9024) [inline conformance_pack/ebs.sp:340]:
```suggestion
        when encrypted then title || ' encryption enabled.'
```

File: conformance_pack/ebs.sp [hunk 26]
```diff
@@ -317,3 +327,22 @@ query "ebs_volume_unused" {
...
+        else title || ' is unencrypted.'
```
Comment (khushboo9024) [inline conformance_pack/ebs.sp:341]:
```suggestion
        else title || ' encryption disabled.'
```

File: conformance_pack/cloudtrail.sp [hunk 27]
```diff
@@ -205,6 +205,16 @@ control "cloudtrail_multi_region_read_write_enabled" {
   })
 }
 
+control "cloudtrail_trail_insight_selectors_and_logging_enabled" {
+  title       = "Ensure that CloudTrail trails should have insight selectors and logging enabled"
```
Comment (khushboo9024) [inline conformance_pack/cloudtrail.sp:209]:
```suggestion
  title       = "CloudTrail trails should have insight selectors and logging enabled"
```

File: conformance_pack/cloudtrail.sp [hunk 28]
```diff
@@ -577,6 +587,29 @@ query "cloudtrail_multi_region_read_write_enabled" {
...
+      arn as resource,
+      case
+        when not is_logging  then 'skip'
+        when is_logging and has_insight_selectors then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when not is_logging  then title || ' logging is disabled.'
+        when is_logging and has_insight_selectors then title || ' has insight selectors and logging is enabled.'
+        else title || ' does not has insight selectors and logging is enabled.'
```
Comment (khushboo9024) [inline conformance_pack/cloudtrail.sp:602]:
```suggestion
        when is_logging and has_insight_selectors then title || ' has insight selectors and logging enabled.'
        else title || ' does not has insight selectors enabled.'
```

File: conformance_pack/backup.sp [hunk 29]
```diff
@@ -186,3 +206,63 @@ query "backup_recovery_point_min_retention_35_days" {
...
+    )  
+    select
+      'arn:' || r.partition || '::' || r.region || ':' || r.account_id as resource,
+      case
+        when cp.count > 0 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when cp.count > 0 then cp.count || ' backup plan(s) exist in ' || r.region || '.'
+        else 'No backup plans exist in ' || r.region || '.'
```
Comment (khushboo9024) [inline conformance_pack/backup.sp:231]:
```suggestion
        when cp.count > 0 then cp.count || ' backup plan(s) exist in region ' || r.region || '.'
        else 'No backup plans exist in ' || r.region || '.'
```

File: conformance_pack/apigateway.sp [hunk 30]
```diff
@@ -109,6 +109,26 @@ control "apigateway_rest_api_authorizers_configured" {
   })
 }
 
+control "apigateway_rest_api_endpoint_restrict_public_access" {
+  title       = "Ensure that API endpoint type in API Gateway is configured to private"
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:113]:
```suggestion
  title       = "API Gateway REST API endpoint type should be configured to private"
```

File: conformance_pack/apigateway.sp [hunk 31]
```diff
@@ -109,6 +109,26 @@ control "apigateway_rest_api_authorizers_configured" {
...
+  description = "This control checks whether API Gateway endpoint is public or private. This rule is non compliant if API Gateway endpoint is public."
+  query       = query.apigateway_rest_api_endpoint_restrict_public_access
+
+  tags = merge(local.conformance_pack_apigateway_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "api_gatewayv2_route_authorizer_enabled" {
+  title       = "Ensure that API Gateway V2 has authorizer configured"
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:123]:
```suggestion
  title       = "API Gateway V2 authorizer should be configured"
```

File: conformance_pack/drs.sp [hunk 32]
```diff
@@ -0,0 +1,48 @@
+locals {
+  conformance_pack_drs_common_tags = merge(local.aws_compliance_common_tags, {
+    service = "AWS/DRS"
+  })
+}
+
+control "drs_job_exists" {
+  title       = "Ensure DRS is enabled with jobs"
```
Comment (khushboo9024) [inline conformance_pack/drs.sp:8]:
```suggestion
  title       = "DRS jobs should be enabled"
```

File: conformance_pack/ec2.sp [hunk 33]
```diff
@@ -269,6 +269,16 @@ control "ec2_instance_virtualization_type_no_paravirtual" {
   })
 }
 
+control "ec2_ami_restrict_public_access" {
+  title       = "Ensure there are no EC2 AMIs set as public"
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:273]:
```suggestion
  title       = "EC2 AMIs should restrict public access"
```

File: conformance_pack/iam.sp [hunk 34]
```diff
@@ -522,6 +522,16 @@ control "iam_policy_unused" {
   })
 }
 
+control "iam_access_analyzer_enabled_without_findings" {
+  title         = "Ensure that IAM Access analyzer is enabled without findings"
```
Comment (khushboo9024) [inline conformance_pack/iam.sp:526]:
```suggestion
  title         = "IAM Access analyzer should be enabled without findings"
```

File: conformance_pack/vpc.sp [hunk 35]
```diff
@@ -333,6 +333,46 @@ control "vpc_security_group_allows_ingress_authorized_ports" {
   })
 }
 
+control "vpc_security_group_allows_ingress_to_cassandra_ports" {
+  title       = "Ensure no security groups allow ingress from 0.0.0.0/0 or ::/0 to Cassandra ports 7199 or 9160 or 8888"
```
Comment (khushboo9024) [inline conformance_pack/vpc.sp:337]:
```suggestion
  title       = VPC security groups should restrict ingress from 0.0.0.0/0 or ::/0 to Cassandra ports 7199 or 9160 or 8888"
```

Comment (khushboo9024) [reply conformance_pack/vpc.sp:337]: Please update all VPC queries title in this format.

**Review by madhushreeray30 — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please have a look at the comments. Thanks!

File: conformance_pack/apigateway.sp [hunk 36]
```diff
@@ -109,6 +109,26 @@ control "apigateway_rest_api_authorizers_configured" {
...
+  query       = query.apigateway_rest_api_endpoint_restrict_public_access
+
+  tags = merge(local.conformance_pack_apigateway_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "api_gatewayv2_route_authorizer_configured" {
+  title       = "API Gateway V2 authorizer should be configured"
+  description = "This control checks whether API Gateway V2 has authorizer configured. This rule is non compliant if API Gateway V2 have no authorizers configured."
```
Comment (misraved) [inline conformance_pack/apigateway.sp:124]:
```suggestion
  description = "This control checks whether API Gateway V2 has an authorizer configured. This rule is non-compliant if API Gateway V2 has no authorizers configured."
```

File: conformance_pack/apigateway.sp [hunk 37]
```diff
@@ -252,6 +272,43 @@ query "apigateway_rest_api_authorizers_configured" {
...
+  sql = <<-EOQ
+    select
+      'arn:' || partition || ':apigateway:' || region || '::/apis/' || api_id || '/routes/' || route_id as resource,
+      case
+        when authorizer_id is null then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when authorizer_id is null then route_id || ' authorizer not configured.'
+        else route_id || ' authorizer configured.'
```
Comment (misraved) [inline conformance_pack/apigateway.sp:304]: Should we add how many or what authorizers are configured for the route IDs?

File: conformance_pack/backup.sp [hunk 38]
```diff
@@ -82,6 +82,26 @@ control "backup_recovery_point_min_retention_35_days" {
   })
 }
 
+control "backup_plan_region_configured" {
+  title       = "Backup plan should exist in a region"
+  description = "Ensures that there exist at least one backup plan in a region. The rule is non-compliant if there exist no backup plan in a region."
```
Comment (misraved) [inline conformance_pack/backup.sp:87]:
```suggestion
  description = "Ensure that there exists at least one backup plan in a region. The rule is non-compliant if there are no backup plans in a region."
```

File: conformance_pack/backup.sp [hunk 39]
```diff
@@ -82,6 +82,26 @@ control "backup_recovery_point_min_retention_35_days" {
...
+  query       = query.backup_plan_region_configured
+
+  tags = merge(local.conformance_pack_backup_common_tags, {
+    other_checks = "true"
+  })
+}
+
+control "backup_vault_region_configured" {
+  title       = "Backup vaults should exist in a region"
+  description = "Ensures that there exist at least one backup vault in a region. The rule is non-compliant if there exist no backup vault in a region."
```
Comment (misraved) [inline conformance_pack/backup.sp:97]:
```suggestion
  description = "Ensure that there exists at least one backup vault in a region. The rule is non-compliant if there are no backup vaults in a region."
```

File: conformance_pack/backup.sp [hunk 40]
```diff
@@ -186,3 +206,63 @@ query "backup_recovery_point_min_retention_35_days" {
...
+    select
+      'arn:' || r.partition || '::' || r.region || ':' || r.account_id as resource,
+      case
+        when cp.count > 0 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when cp.count > 0 then cp.count || ' backup plan(s) exist in region ' || r.region || '.'
+        else 'No backup plans exist in region ' || r.region || '.'
+      end as reason
```
Comment (misraved) [inline conformance_pack/backup.sp:232]: Do we need to add the region information? Won't it be already available in the common dimension?

File: conformance_pack/backup.sp [hunk 41]
```diff
@@ -186,3 +206,63 @@ query "backup_recovery_point_min_retention_35_days" {
...
+    )  
+    select
+      'arn:' || r.partition || '::' || r.region || ':' || r.account_id as resource,
+      case
+        when v.count > 0 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when v.count > 0 then v.count || ' backup vault(s) exist in region ' || r.region || '.'
+        else 'No backup vault exists in region ' || r.region || '.'
```
Comment (misraved) [inline conformance_pack/backup.sp:261]: Do we need to add the region information? Won't the data be automatically available via the region common dimension?

File: conformance_pack/cloudfront.sp [hunk 42]
```diff
@@ -148,6 +148,16 @@ control "cloudfront_distribution_origin_access_identity_enabled" {
   })
 }
 
+control "cloudfront_distribution_field_level_encryption_enabled" {
+  title       = "CloudFront distributions should have field level encryption enabled"
+  description = "This control checks whether an Amazon CloudFront distribution have field level encryption enabled. The control fails if CloudFront distribution field level encryption is not enabled."
```
Comment (misraved) [inline conformance_pack/cloudfront.sp:153]:
```suggestion
  description = "This control checks whether an Amazon CloudFront distribution has field-level encryption enabled. The control fails if CloudFront distribution field-level encryption is not enabled."
```

File: conformance_pack/cloudtrail.sp [hunk 43]
```diff
@@ -577,6 +587,29 @@ query "cloudtrail_multi_region_read_write_enabled" {
   EOQ
 }
 
+query "cloudtrail_trail_insight_selectors_and_logging_enabled" {
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when not is_logging  then 'skip'
```
Comment (misraved) [inline conformance_pack/cloudtrail.sp:595]: Shouldn't the control go to an alarm if logging is disabled?

As per the description - `CloudTrail trails should have insight selectors and logging enabled`, I think we should fail the control if it does not have logging enabled.

File: conformance_pack/ebs.sp [hunk 44]
```diff
@@ -141,6 +141,16 @@ control "ebs_volume_unused" {
   })
 }
 
+control "ebs_snapshot_encryption_enabled" {
+  title       = "EBS snapshots should be encrypted"
+  description = "Ensure if EBS snapshots are encrypted. This rule is non-compliant if the EBS snapshot is not encrypted."
```
Comment (misraved) [inline conformance_pack/ebs.sp:146]:
```suggestion
  description = "Ensure that EBS snapshots are encrypted. This rule is non-compliant if the EBS snapshot is not encrypted."
```

File: conformance_pack/ec2.sp [hunk 45]
```diff
@@ -633,6 +643,25 @@ query "ec2_instance_virtualization_type_no_paravirtual" {
   EOQ
 }
 
+query "ec2_ami_restrict_public_access" {
+  sql = <<-EOQ
+    select
+      'arn:' || partition || '::' || region || ':' || account_id as resource,
```
Comment (misraved) [inline conformance_pack/ec2.sp:649]: Is this a correct `arn` format for an EC2 AMI?

File: conformance_pack/guardduty.sp [hunk 46]
```diff
@@ -50,6 +50,16 @@ control "guardduty_finding_archived" {
   })
 }
 
+control "guardduty_no_high_severity_findings" {
+  title       = "GuardDuty Detector should not have high severity finding"
+  description = "GuardDuty generates a finding whenever it detects unexpected and potentially malicious activity in your AWS environment. If critical findings are not addressed threats can spread in the environment. This rule is non-compliant if there are any high severity findings."
```
Comment (misraved) [inline conformance_pack/guardduty.sp:55]:
```suggestion
  description = "GuardDuty generates a finding whenever it detects unexpected and potentially malicious activity in your AWS environment. If critical findings are not addressed threats can spread in the environment. This rule is non-compliant if there are high severity findings."
```

File: conformance_pack/guardduty.sp [hunk 47]
```diff
@@ -50,6 +50,16 @@ control "guardduty_finding_archived" {
   })
 }
 
+control "guardduty_no_high_severity_findings" {
+  title       = "GuardDuty Detector should not have high severity finding"
```
Comment (misraved) [inline conformance_pack/guardduty.sp:54]:
```suggestion
  title       = "GuardDuty Detector should not have high severity findings"
```

**Review by misraved — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please take a look at the review comments. Thanks!!

**Pull request merged**